rust/onnxruntime/examples/print_structure.rs

```diff
@@ -0,0 +1,38 @@
+//! Display the input and output structure of an ONNX model.
+use onnxruntime::environment;
+use std::error::Error;
+
+fn main() -> Result<(), Box<dyn Error>> {
+    // provide path to .onnx model on disk
+    let path = std::env::args()
+        .nth(1)
+        .expect("Must provide an .onnx file as the first arg");
+
+    let environment = environment::Environment::builder()
+        .with_name("onnx metadata")
+        .with_log_level(onnxruntime::LoggingLevel::Verbose)
+        .build()?;
+
+    let session = environment
+        .new_session_builder()?
+        .with_graph_optimization_level(onnxruntime::GraphOptimizationLevel::Basic)?
+        .with_model_from_file(path)?;
+
+    println!("Inputs:");
+    for (index, input) in session.inputs.iter().enumerate() {
```

Comment:
>session.inputs.iter() [](http://example.com/codeflow?start=26&length=21)

The original interfaces in C/C++ `GetInput/OutputName()` has been leaking memory unless explicitly deallocated. Those interfaces were deprecated in the most recent release. I have not looked at the code generator yet, but has this been taken care of before I forget?

Reply:
I plan on going through this PR, please, have patience.

Reply:
rust/onnxruntime/src/session.rs L 660 of the high-level bindings use the allocator to free the memory. The low-level (auto generated) bindings will have the docstrings from the C api.